pkg/platform/kube/test/platform_test.go

```diff
@@ -762,6 +766,189 @@ func (suite *ProjectTestSuite) TestDelete() {
+	createAPIGatewayOptions := suite.compileCreateAPIGatewayOptions("apigw-to-delete",
+		functionToDeleteA.FunctionConfig.Meta.Name)
+	createAPIGatewayOptions.APIGatewayConfig.Meta.Labels["nuclio.io/project-name"] = projectToDeleteConfig.Meta.Name
+	err = suite.Platform.CreateAPIGateway(createAPIGatewayOptions)
+	suite.Require().NoError(err)
+	defer suite.Platform.DeleteAPIGateway(&platform.DeleteAPIGatewayOptions{ // nolint: errcheck
+		Meta: createAPIGatewayOptions.APIGatewayConfig.Meta,
+	})
+
+	suite.WaitForAPIGatewayState(&platform.GetAPIGatewaysOptions{
+		Name:      createAPIGatewayOptions.APIGatewayConfig.Meta.Name,
+		Namespace: createAPIGatewayOptions.APIGatewayConfig.Meta.Namespace,
+	}, platform.APIGatewayStateReady, 10*time.Second)
+
+	// create 2 function events for function B (deleted along with `projectToDeleteConfig`)
+	functionEventA := suite.CompileCreateFunctionEventOptions("function-event-a",
+		functionToDeleteB.FunctionConfig.Meta.Name)
+	err = suite.Platform.CreateFunctionEvent(functionEventA)
+	suite.Require().NoError(err)
+	defer suite.Platform.DeleteFunctionEvent(&platform.DeleteFunctionEventOptions{ // nolint: errcheck
+		Meta: platform.FunctionEventMeta{
+			Name:      functionEventA.FunctionEventConfig.Meta.Name,
+			Namespace: suite.Namespace,
+		},
+	})
+
+	functionEventB := suite.CompileCreateFunctionEventOptions("function-event-b",
+		functionToDeleteB.FunctionConfig.Meta.Name)
+	err = suite.Platform.CreateFunctionEvent(functionEventB)
+	suite.Require().NoError(err)
+	defer suite.Platform.DeleteFunctionEvent(&platform.DeleteFunctionEventOptions{ // nolint: errcheck
+		Meta: platform.FunctionEventMeta{
+			Name:      functionEventB.FunctionEventConfig.Meta.Name,
+			Namespace: suite.Namespace,
+		},
+	})
+
+	// try restrict - expect it to fail (project has sub resources)
+	err = suite.Platform.DeleteProject(&platform.DeleteProjectOptions{
+		Meta: platform.ProjectMeta{
+			Name:      projectToDeleteConfig.Meta.Name,
+			Namespace: suite.Namespace,
+		},
+		Strategy: platform.DeleteProjectStrategyRestrict,
+	})
+	suite.Require().Error(err)
+
+	// try cascade - expect it succeed
+	err = suite.Platform.DeleteProject(&platform.DeleteProjectOptions{
+		Meta: platform.ProjectMeta{
+			Name:      projectToDeleteConfig.Meta.Name,
+			Namespace: suite.Namespace,
+		},
+		Strategy: platform.DeleteProjectStrategyCascade,
+	})
+	suite.Require().NoError(err)
+
+	// assertion - project should be deleted
+	projects, err := suite.Platform.GetProjects(&platform.GetProjectsOptions{
+		Meta: platform.ProjectMeta{
+			Name:      projectToDeleteConfig.Meta.Name,
+			Namespace: suite.Namespace,
+		},
+	})
+	suite.Require().NoError(err)
+	suite.Require().Len(projects, 0)
+
+	suite.Logger.InfoWith("Ensuring resources were removed (deletion is being executed in background")
+
+	// ensure api gateway deleted
+	err = common.RetryUntilSuccessful(time.Minute, 3*time.Second, func() bool {
```

Comment:
Why not using `WaitForResourcesDeletionCompletion` then no retry until successful is needed ?

Reply:
good point, adding that.

I added `WaitForResourcesDeletionCompletion` only after that test